### PR TITLE
[chrome-api-jstocxxtest] Add tests for list readers request

### PR DESCRIPTION
**Pull request by ViktoriiaKovalova**

### CODE REVIEW

File: smart_card_connector_app/src/chrome-api-jstocxxtest.js [hunk 1]
```diff
@@ -162,6 +162,96 @@ goog.exportSymbol('testChromeApiProviderToCpp', {
...
+    await chrome.smartCardProviderPrivate['reportReleaseContextResult']
+        .$waitAndVerify();
+    mockChromeApi.dispatchEvent(
+        'onListReadersRequested', /*requestId=*/ 125, sCardContext);
+    await chrome.smartCardProviderPrivate['reportListReadersResult']
+        .$waitAndVerify();
+  },
+
+  // Test ListReaders returns a one-item list when there's a single attached device.
+  'testListReaders_singleDevice': async function() {
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-jstocxxtest.js:230]: Maybe one more test with multiple devices?

File: smart_card_connector_app/src/chrome-api-jstocxxtest.js [hunk 2]
```diff
@@ -162,6 +162,96 @@ goog.exportSymbol('testChromeApiProviderToCpp', {
         'onReleaseContextRequested', /*requestId=*/ 42, BAD_CONTEXT);
     await chrome.smartCardProviderPrivate['reportReleaseContextResult']
         .$waitAndVerify();
-  }
+  },
+
+  // Test ListReaders with no readers attached.
+  'testListReaders_none': async function() {
+    let sCardContext = 0;
+    chrome
```
[the remark this answers is not in the document]
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-jstocxxtest.js]: The expectation for establishing context is set in the beginning, but the actual dispatch of establish context request happens in the middle of the test. So only setting up expectations can be factored out into 'setUp', if I understood your suggestion correctly.

I moved the expectations into separate functions, maybe they look less ugly now :)

[the remark this answers is not in the document]
Comment (emaxx-google) [reply smart_card_connector_app/src/chrome-api-jstocxxtest.js]: Ack - the new version looks better. Agreed that fully automating this boilerplate wouldn't be easy.

File: smart_card_connector_app/src/chrome-api-jstocxxtest.js [hunk 3]
```diff
@@ -162,6 +205,111 @@ goog.exportSymbol('testChromeApiProviderToCpp', {
...
+        .$waitAndVerify();
+  },
+
+  // Test that ListReaders requested with already released context returns
+  // INVALID_HANDLE error.
+  'testListReaders_releasedContext': async function() {
+    let establishContextResult = EMPTY_CONTEXT_RESULT;
+    expectReportEstablishContext(
+        /*requestId=*/ 123, 'SUCCESS', establishContextResult);
+    expectReportReleaseContext(/*requestId=*/ 124, 'SUCCESS')
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-jstocxxtest.js:237]: nit: Terminate the statement with a semicolon (ditto in other places below).

**Review by emaxx-google — APPROVED** [review]

File: smart_card_connector_app/src/chrome-api-jstocxxtest.js (hunk 3 above)
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-jstocxxtest.js]: thanks;

**Pull request merged**